src/Evolution/Systems/ScalarAdvection/FiniteDifference/ReconstructWork.hpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim>
+class Direction;
+template <size_t Dim>
+class Element;
+template <size_t Dim>
+class ElementId;
+template <size_t Dim>
+class Mesh;
+template <typename TagsList>
+class Variables;
+namespace evolution::dg::subcell {
+class NeighborData;
+}  // namespace evolution::dg::subcell
+namespace gsl {
+template <typename>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace ScalarAdvection::fd {
+/*!
+ * \brief Reconstructs the scalar field \f$U\f$. All results are written into
+ * `vars_on_lower_face` and `vars_on_upper_face`.
+ *
+ * Need more detailed docs - direction, lower and upper face?
```

Comment:
I would say hold off on writing more until we see what unstructured mesh reconstruction looks like :)